packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -882,7 +870,7 @@ export function interopNamedImports(
     str.overwrite(
       expStart,
       expEnd,
-      `import('${rewrittenUrl}').then(m => (${interopHelper.toString()})(m.default))` +
+      `import('${rewrittenUrl}').then(m => m.default && m.default.__esModule ? m.default : { ...(typeof m.default === 'object' && !Array.isArray(m.default) ? m.default : {}), default: m.default })` +
```

Comment:
I think the function form is shorter, and we can use `?.` during dev
```suggestion
      `import('${rewrittenUrl}').then(m => (d => d?.__esModule ? d : { ...(typeof d === 'object' && !Array.isArray(d) ? d : {}), default: d })(m.default))` +
```

Reply:
> I think the function form is shorter, and we can use `?.` during dev

Looks good!